makeflow/src/makeflow.c

```diff
@@ -1424,6 +1517,15 @@ int main(int argc, char *argv[])
 			remote_jobs_max = MIN(local_jobs_max, n);
 		}
 	}
+	
+	struct dag_node *n;
+	// check to ensure that all jobs can be ran with resources given
+	for(n = d->nodes; n; n = n->next) {
```

Comment:
Move this loop down after the calls to makeflow_check and makeflow_check_batch_consistency. Then surround it with an if statement that checks if we are in clean mode, because we only want to fail out if we are going to run the makeflow and not prevent cleaning.